### ISSUE TITLE
[Gtk3] Window decoration defects with native Gtk dialogs

### ISSUE THREAD

**Issue by Lin-Buo-Ren** [ubuntu/snapcraft-desktop-helpers#150]
Refer the following screenshots:
![2018-08-27 23-00-10](https://user-images.githubusercontent.com/13408130/44667412-4d3f5500-aa4d-11e8-95ec-894077a50f84.png)

![2018-08-27 22-59-48](https://user-images.githubusercontent.com/13408130/44667434-56c8bd00-aa4d-11e8-8b54-46b17ed8e319.png)

Notice:

* The missing shadowing at the edge of the window
* The missing transparency at the corner of the window (rendered as black)

These defects only appear in (seemingly) native Gtk dialogs, control case for comparison:

![2018-08-27 23-00-51](https://user-images.githubusercontent.com/13408130/44667399-41ec2980-aa4d-11e8-87bd-496ef4e98c56.png)

Also not reproduced in XFCE desktop environment, which seemingly doesn't support some GNOME 3 specific features:

![_001](https://user-images.githubusercontent.com/13408130/44667742-35b49c00-aa4e-11e8-9260-6d14b4444cf3.png)

**Comment by oSoMoN** [comment]
That looks similar to https://launchpad.net/bugs/1755119

**Comment by oSoMoN** [comment]
Yeah, not sure either… Can you confirm you're not seeing the issue with the chromium snap?

**Comment by Lin-Buo-Ren** [comment]
It is reproduced in the Chromium snap as well:

![2018-09-05 01-48-16](https://user-images.githubusercontent.com/13408130/45048182-d12cb900-b0ad-11e8-8ae9-a5ce146d2489.png)

**Comment by oSoMoN** [comment]
Indeed, I am seeing the issue again with the chromium snap. Re-opening the bug.
The chromium snap from the candidate/from-source channel is not affected though. An obvious difference is that the snap built from source connects to the gtk-common-themes interfaces (gtk-3-themes, icon-themes, sound-themes).

**Comment by Lin-Buo-Ren** [comment]
This issue is also reproduced on my demo-gtk-common-themes snap, which is a simple snap for testing gtk-common-themes: https://github.com/Lin-Buo-Ren/demo-gtk-common-themes/blob/master/snap/snapcraft.yaml

I noticed that only the Ambiance and Radiance themes are affected, Adwaita isn't.

**Comment by oSoMoN** [comment]
@Lin-Buo-Ren: are you still seeing this issue with your snaps? I just tested again with chromium, and the issue seems to be gone.
